src/main/java/com/bravo/user/converter/PasswordEncryption.java

```diff
@@ -0,0 +1,48 @@
+package com.bravo.user.converter;
+
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+
+import org.apache.commons.codec.binary.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class PasswordEncryption {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PasswordEncryption.class);
+
+  private Cipher encryptionCipher;
+  private Cipher decriptionCipher;
+  
+  @Autowired
+  public PasswordEncryption(@Qualifier("encryptionCipher") Cipher encryptionCipher,
+      @Qualifier("decriptionCipher") Cipher decriptionCipher) {
+    this.encryptionCipher = encryptionCipher;
+    this.decriptionCipher = decriptionCipher;
+  }
+
+  public String encrypt(String rawPassword) {
+    try {
+      return new String(Base64.getEncoder().encode(encryptionCipher.doFinal(rawPassword.getBytes())));
+    } catch (Exception e) {
+      LOGGER.error("Failed to encrypy DB Password");
+      throw new RuntimeException(e);
+    }
+  }
+
+  public boolean matches(String rawPassword, String encryptedPassword) {
+    try {
+      String decodedPassword = new String(decriptionCipher.doFinal(Base64.getDecoder().decode(encryptedPassword)));
+      return StringUtils.equals(rawPassword, decodedPassword);
+    } catch (Exception e) {
```

Comment:
Was Exception the only option on this catch block?

Reply:
I suppose this is a habit I picked up at my current job. 

No, this wasn't the only option. IIRC think there was 4-6 different exceptions that could be thrown on line 40. I know that I wanted to treat all exceptions thrown by that line as I did in this catch block so that gave me two options. Declare all of them and hope that Eclipse caught all relevant ones or just do a catch all. I opted for the latter. 

If there had been more specific actions that needed to be taken in some exception cases, but not all of them that would have been a better argument in my mind to declare the exception types explicitly.